### DISCUSSION TITLE
ValueError during validation using any generic dataset

### DISCUSSION THREAD

**Discussion by a-nolte** [neuralhydrology/neuralhydrology#115] [Q&A] [answered]
During the validation phase with generic datasets, the ValueError "One or more of the specified variables cannot be found in this dataset" keeps popping up for me. The error is related to a part of "tester.py", more precisely here: "obs = xr.isel(time_step=slice(-frequency_factor, None))". To rule out that my dataset is preprocessed wrong, I also used the sample dataset provided in this discussion (https://github.com/neuralhydrology/neuralhydrology/discussions/113) (I used temperature to predict total_precipitation for testing). Unfortunately, the error appears here as well. Could you point me to possible error sources or alternatively provide a working config.yml file for the generic sample dataset? That would be awesome! :)

**Comment by gauchm** [comment, marked as answer]
I think the reason might be that you are using an outdated xarray version. You can check your version via `conda env export`. Could you please try to set up a fresh conda environment with the environment yaml files provided [here](https://github.com/neuralhydrology/neuralhydrology/tree/master/environments), making sure that the xarray version this installs is reasonably recent (e.g., I use v2023.1.0 without problems)?

**Reply by a-nolte** [reply]
Great, thank you! Running my code in a newly installed environment using the yaml file worked for me.
